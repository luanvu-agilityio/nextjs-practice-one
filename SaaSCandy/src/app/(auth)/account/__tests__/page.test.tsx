import { render } from '@testing-library/react';
import AccountPage from '@/app/(auth)/account/page';

// Mock the component
jest.mock('@/components/pages', () => ({
  AccountPageContent: () => (
    <div data-testid='account-page-content'>Account Page Content</div>
  ),
}));

describe('AccountPage - Snapshot Tests', () => {
  it('should match snapshot', () => {
    const { container } = render(<AccountPage />);
    expect(container).toMatchSnapshot();
  });

  it('should render AccountPageContent component', () => {
    const { getByTestId } = render(<AccountPage />);
    expect(getByTestId('account-page-content')).toBeInTheDocument();
  });
});
