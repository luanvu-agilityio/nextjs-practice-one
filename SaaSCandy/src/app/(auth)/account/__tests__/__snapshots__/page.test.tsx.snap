// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`AccountPage - Snapshot Tests should match snapshot 1`] = `
<div>
  <div
    data-testid="account-page-content"
  >
    Account Page Content
  </div>
</div>
`;
